process/elasticproc/tokens/sovereignIndexTokensHandler.go

```diff
@@ -29,73 +22,44 @@ type sovereignIndexTokensHandler struct {
 }
 
 // NewSovereignIndexTokensHandler creates a new sovereign index tokens handler
-func NewSovereignIndexTokensHandler(mainChainElastic factory.ElasticConfig, esdtPrefix string) (*sovereignIndexTokensHandler, error) {
-	var mainChainElasticClient elasticproc.DatabaseClientHandler
-	if mainChainElastic.Enabled {
-		var err error
-		argsEsClient := elasticsearch.Config{
-			Addresses:     []string{mainChainElastic.Url},
-			Username:      mainChainElastic.UserName,
-			Password:      mainChainElastic.Password,
-			Logger:        &logging.CustomLogger{},
-			RetryOnStatus: []int{http.StatusConflict},
-			RetryBackoff:  retryBackOff,
-		}
-		mainChainElasticClient, err = client.NewElasticClient(argsEsClient)
-		if err != nil {
-			return nil, err
-		}
-	}
-
+func NewSovereignIndexTokensHandler(indexingEnabled bool, mainChainElasticClient elasticproc.DatabaseClientHandler, esdtPrefix string) (*sovereignIndexTokensHandler, error) {
```

Comment:
Please do not use a `bool` flag for enabled components in constructor, this is not clean code at all.
You can do something similar to our outportDriver

Create a new interface, which embeds `elasticproc.DatabaseClientHandler` and extends it with an extra function `IsEnabled()` and use that one instead of passing here this bool.